recipes/opencv/4.x/conanfile.py

```diff
@@ -254,6 +309,11 @@ def _configure_cmake(self):
         if self.options.parallel:
             self._cmake.definitions["WITH_TBB"] = self.options.parallel == "tbb"
             self._cmake.definitions["WITH_OPENMP"] = self.options.parallel == "openmp"
+        self._cmake.definitions["WITH_CUDA"] = self.options.with_cuda
+        if self.options.with_cuda:
+            # This allows compilation on older GCC/NVCC, otherwise build errors.
+            self._cmake.definitions["CUDA_NVCC_FLAGS"] = "--expt-relaxed-constexpr"
```

Comment:
Is this required? Or can we set a minimum compiler requirement?

Reply:
I'm not sure if that is necessary. I used to build older versions of OpenCV using [this deprecated recipe](https://github.com/conan-community/conan-opencv) and I copied this line from it. 

[According to the CUDA Toolkit documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#constexpr-functions), this flag is experimental, therefore we cannot set a minimum compiler requirement.

Reply:
Since it's only one compiler I think thats okay